packages/kit/src/runtime/server/page/render.js

```diff
@@ -322,11 +322,17 @@ export async function render_response({
 						})`);
 
 			properties.push(`resolve: ({ id, data, error }) => {
-							const { fulfil, reject } = deferred.get(id);
-							deferred.delete(id);
-
-							if (error) reject(error);
-							else fulfil(data);
+							const try_to_resolve = () => {
+								if (!deferred.has(id)) {
+									setTimeout(try_to_resolve, 0);
```

Comment:
was is necessary after all due to the dynamic imports? If so we should add a comment here stating why this is necessary.

Reply:
this feels suuuuuper hacky, surely we can find another spproach?